### PR TITLE
Assignment 4 Diego Torrelles

### PR DESCRIPTION
**Pull request by diegotorrelles**

### CODE REVIEW

File: Assignment4/course_materials/DiegoTorrelles-190144/task06.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+# Visualize the results
+for s, p, o in g:
+  print(s,p,o)
+
+"""**TASK 6.5: Add UPM as the university where John Smith works**"""
+
+# TO DO
+g.add((ns.UPM, RDF.type, ns.University))
+g.add((ns.JohnSmith, vcard.Works, ns.UPM))
```
Comment (dgarijo) [inline Assignment4/course_materials/DiegoTorrelles-190144/task06.py:76]: This is not a property in vcard. Search for the right one

File: Assignment4/course_materials/DiegoTorrelles-190144/task06.py [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+# Visualize the results
+for s, p, o in g:
+  print(s,p,o)
```
Comment (dgarijo) [inline Assignment4/course_materials/DiegoTorrelles-190144/task06.py:79]: one exercise is missing here

File: Assignment4/course_materials/DiegoTorrelles-190144/task07.py [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+# TO DO
+from rdflib.plugins.sparql import prepareQuery
+
+
+VCARD = Namespace("http://www.w3.org/2001/vcard-rdf/3.0#")
+NS = Namespace("http://somewhere#")
+
+q1 = prepareQuery('''
+  SELECT ?subClass WHERE {
+    ?subClass rdfs:subClassOf ns:LivingThing.
```
Comment (dgarijo) [inline Assignment4/course_materials/DiegoTorrelles-190144/task07.py:35]: this does not retrieve the subclasses of subclasses of Living thing. Use chained properties

File: Assignment4/course_materials/DiegoTorrelles-190144/task07.py [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+
+# TO DO
+q2 = prepareQuery('''
+  SELECT
+    ?Subject
+  WHERE {
+    {
+      ?Subject rdf:type ns:Person.
+    } UNION{
+      ?subClass rdfs:subClassOf ns:Person.
```
Comment (dgarijo) [inline Assignment4/course_materials/DiegoTorrelles-190144/task07.py:56]: same issue as above

File: Assignment4/course_materials/DiegoTorrelles-190144/task07.py [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+      ?Subject rdf:type ?subClass.
+      ?Subject ?Properties ?Class.
+    }
+  }
+  ''',
+  initNs = { "rdfs":RDFS, "ns":NS, "rdf":RDF}
+)
+
```
Comment (dgarijo) [inline Assignment4/course_materials/DiegoTorrelles-190144/task07.py:89]: missing 7.4, 7.5 and ALL the exercises in rdflib

**Review by dgarijo — CHANGES_REQUESTED** [review]
Missing many exercises

**Comment by anaigmo** [comment]
Submitted and mostly fixed, no more PRs allowed. The submission does not follow the instruction, so it will not be merged

**Pull request closed without merging**